plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeWriter.java

```diff
@@ -198,7 +198,7 @@ private static DeltaLakeJsonFileStatistics readStatistics(
             typeForColumn.put(dataColumnNames.get(i), dataColumnTypes.get(i));
         }
 
-        TrinoInputFile inputFile = fileSystem.newInputFile(new Path(tableLocation, relativeFilePath).toString());
+        TrinoInputFile inputFile = fileSystem.newInputFile(stringPathOf(tableLocation, relativeFilePath));
```

Comment:
I'm asking naively, why is not enough something like this:

```
new Path(tableLocation, relativeFilePath).toString
```

?

Reply:
> I'm asking naively, why is not enough something like this:
> 
> ```
> new Path(tableLocation, relativeFilePath).toString
> ```
> 
> ?

The idea behind this PR is to remove dependency to hadoop and migrate to TrinoFileSystem as discribed in https://github.com/trinodb/trino/issues/15921

Reply:
Sorry for the confusion. I was thinking that `java.nio.file.Path` was being used in `trino-iceberg` , but  I see that it still uses hadoop `Path`.